src/AddImageChecks.cpp

```diff
@@ -93,6 +94,51 @@ class FindBuffers : public IRGraphVisitor {
     }
 };
 
+class TrimStmtToPartsThatAccessBuffers : public IRMutator {
+    bool touches_buffer = false;
+    const map<string, FindBuffers::Result> &buffers;
+
+    using IRMutator::visit;
+
+    Expr visit(const Call *op) override {
+        touches_buffer |= (buffers.find(op->name) != buffers.end());
+        return IRMutator::visit(op);
+    }
+    Stmt visit(const Provide *op) override {
+        touches_buffer |= (buffers.find(op->name) != buffers.end());
+        return IRMutator::visit(op);
+    }
+    Expr visit(const Variable *op) override {
+        if (op->param.defined() && op->param.is_buffer()) {
+            touches_buffer |= (buffers.find(op->param.name()) != buffers.end());
+        }
+        return IRMutator::visit(op);
+    }
+
+    Stmt visit(const Block *op) override {
+        bool old_touches_buffer = touches_buffer;
```

Comment:
Is `ScopedValue<bool>` not a good option here?

Reply:
I don't think so. The function mutates the value of old_touches_buffer as it goes - it doesn't just restore it at the end.